shell/edit/provisioning.cattle.io.cluster/rke2.vue

```diff
@@ -1061,6 +1066,20 @@ export default {
     nlToBr,
     set,
 
+    /**
+     * set instanceNameLimit to 15 to all pool machine if truncateHostnames checkbox is clicked
+     */
+    truncateName() {
+      if (this.truncateHostnames) {
```

Comment:
This is incorrect - we no longer have to set on each pool.

We should now be setting `machinePoolDefaults.hostnameLengthLimit` on the cluster.

Note that `machinePoolDefaults` may not exist, so we need to add that object first.

if truncation not set, we should remove the `machinePoolDefaults` object if it is empty